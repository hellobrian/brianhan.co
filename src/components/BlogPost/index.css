/* src/components/StyledLink */
.blog a {
  display: inline-block;
  text-decoration: none;
  color: currentColor;
  background-repeat: no-repeat;
  background-image: var(--gradient--link);
  background-size: 100% 0.375rem;
  background-position: 0 115%;
}

.blog p {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.blog h1,
.blog h2,
.blog h3,
.blog h4,
.blog h5,
.blog h6 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.blog blockquote {
  font-style: italic;
  padding: 1rem;
  border-top-right-radius: 10px;
  margin-left: 0.5rem;
  border-left: 5px solid var(--salmon);
  background-color: var(--salmon--light--rgba50);
}
