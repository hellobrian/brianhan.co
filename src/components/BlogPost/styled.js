import styled from 'styled-components';
import { COMMON_BREAKPOINTS } from 'src/utils';

export const HomeLink = styled.p`
  font-family: 'Karla', sans-serif;
  font-size: 1.5rem;
  font-weight: 700;
  line-height: 1.1;
  margin-bottom: 1rem;
  padding-left: 0.5rem;

  @media screen and (min-width: 600px) {
    padding-left: 0;
  }

  @media screen and (min-width: 800px) {
    font-size: 2rem;
  }

  ${COMMON_BREAKPOINTS};
`;

export const Title = styled.h1`
  font-size: 2rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  width: 100%;

  @media screen and (min-width: 600px) {
    padding-left: 0;
    padding-right: 0;
  }
  @media screen and (min-width: 800px) {
    font-size: 2.5rem;
  }

  ${COMMON_BREAKPOINTS};
`;

export const SubTitle = styled.p`
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-family: 'Karla', sans-serif;
  font-size: 1.25rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  width: 100%;

  @media screen and (min-width: 600px) {
    padding-left: 0;
    padding-right: 0;
  }

  @media screen and (min-width: 800px) {
    font-size: 1.25rem;
  }

  ${COMMON_BREAKPOINTS};
`;

export const Content = styled.div`
  width: 100%;

  span.gatsby-resp-image-wrapper {
    width: 100vw;
    left: -0.5rem;
  }

  ${COMMON_BREAKPOINTS}
`;
