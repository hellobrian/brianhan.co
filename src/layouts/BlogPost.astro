---
import type { CollectionEntry } from "astro:content";
import BaseHead from "../components/BaseHead.astro";
import Default from "./Default.astro";

import "../styles/blog-post.css";

type Props = CollectionEntry<"blog">["data"];

const {
  title,
  description,
  pubDate,
  updated,
  heroImage,
  heroImageCredit,
  heroImageUrl,
} = Astro.props;
---

<Default title={title} description={description}>
  <main>
    <article class="blog">
      {
        heroImage && (
          <>
            <img width={720} height={360} src={heroImage} alt="" />
            <small style="font-family: Karla, sans-serif;">
              Photo by {heroImageCredit} (<a href={heroImageUrl}>Unsplash</a>)
            </small>
          </>
        )
      }
      <h1 class="title">{title}</h1>
      {pubDate && <time>{pubDate.toLocaleDateString("en-US")}</time>}
      {
        updated && (
          <div>
            Last updated on <time>{updated.toLocaleDateString("en-US")}</time>
          </div>
        )
      }
      <hr />
      <slot />
    </article>
  </main>
</Default>

<style>
  .title {
    font-size: 2em;
    margin: 0.25em 0 0;
  }
  hr {
    border-top: 1px solid #ddd;
    margin: 1rem 0;
  }
</style>
