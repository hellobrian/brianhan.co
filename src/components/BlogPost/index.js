import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';
import { HomeLink, Title, SubTitle, Content } from './styled';
import './index.css';

export const text = {
  name: `Brian Han`,
};

const BlogPost = ({ title, subtitle, html }) => (
  <div className="blog">
    <HomeLink>
      <Link to="/">{text.name}</Link>
    </HomeLink>
    <Title>{title}</Title>
    {subtitle && <SubTitle>{subtitle}</SubTitle>}
    <Content dangerouslySetInnerHTML={{ __html: html }} />
  </div>
);

BlogPost.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  html: PropTypes.string.isRequired,
};

export default BlogPost;
