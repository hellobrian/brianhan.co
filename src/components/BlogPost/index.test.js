import React from 'react';
import { render } from 'react-testing-library';
import BlogPost from './index';

describe('BlogPost', () => {
  it('renders correctly', () => {
    const { container } = render(
      <BlogPost title="title" subtitle="subtitle" html="<div>html</div>" />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
