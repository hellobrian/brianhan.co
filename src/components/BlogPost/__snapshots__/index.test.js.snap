// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPost renders correctly 1`] = `
.c0 {
  font-family: 'Karla',sans-serif;
  font-size: 1.5rem;
  font-weight: 700;
  line-height: 1.1;
  margin-bottom: 1rem;
  padding-left: 0.5rem;
}

.c1 {
  font-size: 2rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  width: 100%;
}

.c2 {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-family: 'Karla',sans-serif;
  font-size: 1.25rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  width: 100%;
}

.c3 {
  width: 100%;
}

@media screen and (min-width:600px) {
  .c0 {
    padding-left: 0;
  }
}

@media screen and (min-width:800px) {
  .c0 {
    font-size: 2rem;
  }
}

@media screen and (min-width:600px) {
  .c0 {
    margin-left: auto;
    margin-right: auto;
    width: 500px;
  }
}

@media screen and (min-width:800px) {
  .c0 {
    width: 600px;
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    width: 700px;
  }
}

@media screen and (min-width:600px) {
  .c1 {
    padding-left: 0;
    padding-right: 0;
  }
}

@media screen and (min-width:800px) {
  .c1 {
    font-size: 2.5rem;
  }
}

@media screen and (min-width:600px) {
  .c1 {
    margin-left: auto;
    margin-right: auto;
    width: 500px;
  }
}

@media screen and (min-width:800px) {
  .c1 {
    width: 600px;
  }
}

@media screen and (min-width:1200px) {
  .c1 {
    width: 700px;
  }
}

@media screen and (min-width:600px) {
  .c2 {
    padding-left: 0;
    padding-right: 0;
  }
}

@media screen and (min-width:800px) {
  .c2 {
    font-size: 1.25rem;
  }
}

@media screen and (min-width:600px) {
  .c2 {
    margin-left: auto;
    margin-right: auto;
    width: 500px;
  }
}

@media screen and (min-width:800px) {
  .c2 {
    width: 600px;
  }
}

@media screen and (min-width:1200px) {
  .c2 {
    width: 700px;
  }
}

@media screen and (min-width:600px) {
  .c3 {
    margin-left: auto;
    margin-right: auto;
    width: 500px;
  }
}

@media screen and (min-width:800px) {
  .c3 {
    width: 600px;
  }
}

@media screen and (min-width:1200px) {
  .c3 {
    width: 700px;
  }
}

<div
  class="blog"
>
  <p
    class="c0"
  >
    <a
      href="/"
    >
      Brian Han
    </a>
  </p>
  <h1
    class="c1"
  >
    title
  </h1>
  <p
    class="c2"
  >
    subtitle
  </p>
  <div
    class="c3"
  >
    <div>
      html
    </div>
  </div>
</div>
`;
